Prefetch unit tests are flaky

**Describe the bug**
The prefetcher unit tests randomly fail with the error: `spans digest do not match`. 

**Steps To Reproduce**
- `cd` into `fs`

- Run the tests multiple times.

```
$ go clean -testcache && go test ./...

ok      github.com/awslabs/soci-snapshotter/fs    0.009s
?       github.com/awslabs/soci-snapshotter/fs/config    [no test files]
ok      github.com/awslabs/soci-snapshotter/fs/layer    8.559s
?       github.com/awslabs/soci-snapshotter/fs/metrics/common    [no test files]
?       github.com/awslabs/soci-snapshotter/fs/metrics/layer    [no test files]
ok      github.com/awslabs/soci-snapshotter/fs/reader    3.878s
ok      github.com/awslabs/soci-snapshotter/fs/remote    12.936s
?       github.com/awslabs/soci-snapshotter/fs/source    [no test files]
ok      github.com/awslabs/soci-snapshotter/fs/span-manager    0.629s

$ go clean -testcache && go test ./...
ok      github.com/awslabs/soci-snapshotter/fs    0.008s
?       github.com/awslabs/soci-snapshotter/fs/config    [no test files]
ok      github.com/awslabs/soci-snapshotter/fs/layer    8.671s
?       github.com/awslabs/soci-snapshotter/fs/metrics/common    [no test files]
?       github.com/awslabs/soci-snapshotter/fs/metrics/layer    [no test files]
ok      github.com/awslabs/soci-snapshotter/fs/reader    3.910s
ok      github.com/awslabs/soci-snapshotter/fs/remote    12.939s
?       github.com/awslabs/soci-snapshotter/fs/source    [no test files]
--- FAIL: TestStateTransition (0.00s)
    --- FAIL: TestStateTransition/max_span_-_prefetch (0.00s)
        span_manager_test.go:211: failed resolving the span for prefetch
    --- FAIL: TestStateTransition/max_span_-_on_demand_fetch (0.00s)
        span_manager_test.go:220: failed getting the span for on-demand fetch
FAIL
FAIL    github.com/awslabs/soci-snapshotter/fs/span-manager    0.631s
FAIL
```




**Expected behavior**
The prefetcher unit tests pass consistently. 



**Configuration (please complete the following information):**
 - OS: N/A
 - Snapshotter Version: Commit ID #714f1a971e6
 - Containerd Version: N/A

--- a/Makefile
+++ b/Makefile
@@ -74,7 +74,7 @@ check-ltag:
 
 # the very first auto-commit doesn't have a DCO and the first real commit has a slightly different format. Exclude those when doing the check.
 check-dco:
-	$(shell go env GOPATH)/bin/git-validation -run DCO -range 1374574271f4f14126c1d33735339f765f44f0a0..HEAD
+	$(shell go env GOPATH)/bin/git-validation -run DCO -range c5989e95ccd8dede6f39c7197a9db97131ac257b..HEAD
 
 install-check-tools:
 	@curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(shell go env GOPATH)/bin v1.45.2

--- a/c/indexer.c
+++ b/c/indexer.c
@@ -590,6 +590,7 @@ struct gzip_index* blob_to_index(void* buf)
         cur += 8;
         memcpy(&pt->out, cur, 8);
         cur += 8;
+        memset(&pt->bits, 0, sizeof(int));
         memcpy(&pt->bits, cur, 1);
         cur += 1;
         memcpy(&pt->window, cur, WINSIZE);

--- a/fs/span-manager/span_manager_test.go
+++ b/fs/span-manager/span_manager_test.go
@@ -224,16 +224,16 @@ func TestStateTransition(t *testing.T) {
 			if tc.isPrefetch {
 				err := m.ResolveSpan(tc.spanID, r)
 				if err != nil {
-					t.Fatalf("failed resolving the span for prefetch")
+					t.Fatalf("failed resolving the span for prefetch: %v", err)
 				}
 				state := s.state.Load().(spanState)
 				if state != fetched {
 					t.Fatalf("failed transitioning to Fetched state")
 				}
 			} else {
-				_, err := m.GetSpanContent(tc.spanID, 0, s.endUncompOffset, s.endUncompOffset-s.startUncompOffset)
+				_, err := m.GetSpanContent(tc.spanID, 0, s.endUncompOffset-s.startUncompOffset, s.endUncompOffset-s.startUncompOffset)
 				if err != nil {
-					t.Fatalf("failed getting the span for on-demand fetch")
+					t.Fatalf("failed getting the span for on-demand fetch: %v", err)
 				}
 				state := s.state.Load().(spanState)
 				if state != uncompressed {

